src/metax_api/tests/api/rpc/base/views/statistic_rpc.py

```diff
@@ -678,3 +693,89 @@ def test_all_datasets_cumulative_latest_legacy(self):
         self.assertEqual(not_leg_not_lat[-1]['ida_byte_size'],              march_size - legacy_size, not_leg_not_lat)
         self.assertEqual(not_leg_not_lat[-1]['count_cumulative'],           total_count - legacy_count, not_leg_not_lat)
         self.assertEqual(not_leg_not_lat[-1]['ida_byte_size_cumulative'],   total_size - legacy_size, not_leg_not_lat)
+
+
+class StatisticRPCforDrafts(StatisticRPCCommon, CatalogRecordApiWriteCommon):
+    """
+    Tests that drafts are not taken into account when calculating statistics
+    """
+    def test_count_datasets_api_for_drafts(self):
+        """
+        Tests that rpc/statistics/count_datasets returns only count of published datasets
+        """
+        response_1 = self.client.get('/rpc/statistics/count_datasets').data
+
+        self._set_dataset_as_draft(1)
+        self.assertEqual(CatalogRecord.objects.get(pk=1).state, 'draft', CatalogRecord.objects.get(pk=1).id)
+
+        response_2 = self.client.get('/rpc/statistics/count_datasets').data
+        self.assertNotEqual(response_1['count'], response_2['count'],
+        'Drafts should not be returned in count_datasets api')
+
+    def test_all_datasets_cumulative_for_drafts(self):
+        """
+        Tests that /rpc/statistics/all_datasets_cumulative returns only published datasets
+        """
+        url = '/rpc/statistics/all_datasets_cumulative?from_date=2019-06&to_date=2019-06'
+
+        self._set_dataset_creation_date(1, '2019-06-15')
+        response_1 = self.client.get(f'{url}').data
```

Comment:
``f'{url}'`` is a silly way to use the variable url, f-strings should be used if some variable value should be formatted inside a string, but if the variable is a string and there is no other content in the target string, then its better to use the variable directly, e.g. ``self.client.get(url).data``.

Reply:
ok thanks for noting